couchdb/tests/server/views.py

```diff
@@ -118,9 +118,103 @@ def test_return_nothing(self):
         views.map_doc(self.server, doc)
 
 
+class ReduceTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.server = MockQueryServer()
+
+    def test_reduce(self):
+        """should reduce map function result"""
+        state.add_fun(
+            self.server,
+            'def mapfun(doc):\n'
+            '  return ([doc["_id"], i] for i in range(10))'
+        )
+        result = views.map_doc(self.server, {'_id': 'foo'})
+        rresult = views.reduce(
+            self.server,
+            ['def reducefun(keys, values): return sum(values)'],
+            result[0]
+        )
+        self.assertEqual(rresult, [True, [45]])
+
+    def test_reduce_by_many_functions(self):
+        """should proceed map keys-values result by multiple reduce functions"""
+        state.add_fun(
+            self.server,
+            'def mapfun(doc):\n'
+            '  return ([doc["_id"], i] for i in range(10))'
+        )
+        result = views.map_doc(self.server, {'_id': 'foo'})
+        rresult = views.reduce(
+            self.server,
+            ['def reducefun(keys, values): return sum(values)',
+             'def reducefun(keys, values): return max(values)',
+             'def reducefun(keys, values): return min(values)'],
+            result[0]
+        )
+        self.assertEqual(rresult, [True, [45, 9, 0]])
```

Comment:
Good question (: 
By quick looking into the sources multiple reduce functions can occur if you have several views within a single ddoc with a similar map functions but different reduce ones. In this case, similar map functions will produce similar results which can be reduced by a multiple reducers with a single shot.